src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2552,12 +2552,12 @@ void java_lang_Throwable::fill_in_stack_trace(Handle throwable, const methodHand
     return;
   }
 
-  PRESERVE_EXCEPTION_MARK;
-
   JavaThread* thread = JavaThread::active();
+  PreserveExceptionMark pm(thread);
```

Comment:
The result of JavaThread::active() is not necessarily the current thread. So either the new code is wrong or else we should just be using JavaThread::current()

Reply:
My first iteration of this change knew there was a difference then I cleaned it up.  I'll restore the Thread::current() in the PEM constructor.

Reply:
annoyingly every caller of fill_in_stack_trace has the current thread.